src/formatDate.js

```diff
@@ -8,7 +8,71 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldDate = date.split(fromFormat[fromFormat.length - 1]);
+  const newDate = [];
+  let day = '';
+  let month = '';
+  let year = '';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'DD': {
+        day = oldDate[i];
+
+        break;
+      }
+
+      case 'MM': {
+        month = oldDate[i];
+
+        break;
+      }
+
+      default: {
+        year = oldDate[i];
+
+        break;
+      }
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i]) {
+      case 'DD': {
+        newDate.push(day);
+
+        break;
+      }
+
+      case 'MM': {
+        newDate.push(month);
+
+        break;
+      }
+
+      case 'YY': {
+        if (year.length === 4) {
+          newDate.push(year.slice(2));
+        } else {
+          newDate.push(year);
+        }
+
+        break;
+      }
+
+      case 'YYYY': {
+        if (year.length === 4) {
+          newDate.push(year);
+        } else if (year.length === 2 && +year < 30) {
+          newDate.push(`20${year}`);
+        } else {
+          newDate.push(`19${year}`);
+        }
+      }
```

Comment:
Remember to include a `break` statement at the end of each `case` block in your `switch` statements to prevent accidental fall-through. Even though in this case it doesn't cause a bug, it's a good practice to always include it for consistency and to avoid potential future errors.